modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1769,7 +1770,138 @@ def sort_by(
         PandasDataframe
             A new PandasDataframe sorted into lexicographical order by the specified column(s).
         """
-        pass
+        columns = columns if isinstance(columns, list) else [columns]
+        axis = Axis(axis)
+        # The first if selects cases where we are either sorting by a single column that is not
+        # the index, or we are sorting by multiple columns, since in this case, we will have to
+        # shuffle the data before sorting.
+        # The elif selects the case where we are only sorting by index. All partitions maintain
+        # a copy of the index for their data, so if we are just sorting by index, we can make
+        # column partitions, and sort each by their index in parallel, which should be faster than
+        # shuffling our data and sorting.
+        if axis == Axis.ROW_WISE and not (
+            len(columns) == 1 and columns[0] == self.index.names[0]
+        ):
+
+            def sample_func(df):
+                """
+                Sample the given partition.
+
```

Comment:
Is this a variation of samplesort?

Reply:
I believe so - my inspiration was a distributed quick sort, but as per our discussion yesterday, it definitely borrows heavily from sample sort as well!